### DISCUSSION TITLE
Link in readme.md doesn't work anymore

### DISCUSSION THREAD

**Discussion by orcular-org** [community/community#70664] [Repositories] [answered]
### Select Topic Area

Question

### Body

There is a problem with the README.md in my repository: https://github.com/orcular-org/Basislernmodule-Fab-Labs

The readme is in German. There is a link to an English version at the top of the document. It is linked to the file "EN_Readme.md" in the "translations" folder. **This link doesn't work.**

This problem only occurs if you click on the link directly in the repository's start page (in the readme preview below). 

But when you open the README.md [here](https://github.com/orcular-org/Basislernmodule-Fab-Labs/blob/main/README.md) and click the "English" link there, it works.

What is wrong?

**Comment by Janani-Balasooriya** [comment]
Hi @orcular-org , 

Please try using the below link to link to the English version at the top of the document. It will work. I have forked your repository and tried. (Here is the link to forked repo: https://github.com/Janani-Balasooriya/Basislernmodule-Fab-Labs)

https://github.com/orcular-org/Basislernmodule-Fab-Labs/blob/bb2a67d3d2738899adf586a66dcdec32c2fc4c79/translations/EN_Readme.md

Thank you!

**Comment by seyLu** [comment]
#### Hi, @orcular-org,

This is because the link to the en readme is relative to the current page:

![image](https://github.com/community/community/assets/98249191/434f651e-a48d-433a-80dd-bb287d71647b)

<br>

What that means is that it will resolve the links as:

![image](https://github.com/community/community/assets/98249191/a11188b4-ab96-4c98-961a-f06884d3174e)

<br>

But there's no such file in GitHub as GitHub prepends `/blob/<branch>` for files and `/tree/<branch>` for directories:

![image](https://github.com/community/community/assets/98249191/771477f0-fc4c-4b1a-b45e-f66bc644a48c)

<br>

Just prepend a forward slash to resolve this issue:

```
[English](/translations/EN_Readme.md)
```

<br>

Example of this in action:

![image](https://github.com/community/community/assets/98249191/a7562b45-5755-4944-ade3-304c02d94fdf)
![image](https://github.com/community/community/assets/98249191/4a1d92a5-d037-4910-84d9-e438d08f7e6c)

**Reply by nbouvrette** [reply]
I tried your solution as explained [here](https://github.com/orgs/community/discussions/70735) and it does not work.

I am pretty sure this issue is a new Github bug.

**Reply by acusti** [reply, marked as answer]
@nbouvrette this does look like a recently introduced bug. this is a duplicate of https://github.com/orgs/community/discussions/67750, and in that discussion, @SenaJp [pointed out](https://github.com/orgs/community/discussions/67750#discussioncomment-7062910) that the relative links work if they aren’t in a blockquote or list or other non-paragraph markdown block type.

there are two workarounds that i know of, neither of which is ideal:

1. remove the linked text from any non-paragraph markdown block, e.g. remove the `> ` and the `- ` from in front of the text containing the broken link in @orcular-org’s case
2. update the link to have a leading slash and to include the repo’s `/user/repo-name/blob/main/` prefix. so @orcular-org’s link would be `/orcular-org/Basislernmodule-Fab-Labs/blob/main/translations/EN_Readme.md`. note that this solution works out-of-the-box in my testing (both at the root of the repo and in the file browser view) but may break in the future if GitHub fixes this bug, so 🤷

to summarize the bug: relative links are broken for the repo’s root readme file as rendered at the repo homepage if they are contained within a non-paragraph markdown block.

**Reply by orcular-org** [reply]
As you suggested, I removed the link from the quote and it worked! I preferred this solution as I would like to keep all links relative.

It does indeed seem to be a strange bug.

Thanks @acusti and thanks to all the others for trying to help!

**Comment by Dwieght** [comment]
From Chat GPT
The issue you're describing where the link to the English version of your README.md works when opened within the README itself but not when accessed directly from the repository's start page may be due to differences in the relative path of the link.

Here's a possible explanation and how to fix it:

Relative Path Issue: When you create a link to a file within a README, the path can be relative to the location of the README file. However, when accessed from different locations (e.g., the start page of the repository), the relative path might not resolve correctly.

Solution: To ensure the link works from both the README and the repository's start page, you should use an absolute path or a path relative to the repository's root.

Here's how you can modify the link in your README.md:

Original link in your README.md:

[English](translations/EN_Readme.md)

Modified link using an absolute path (assuming the file is in the root of your repository):

[English](translations/EN_Readme.md)
Using an absolute path ensures that the link works consistently, regardless of where it is accessed within the repository or on the start page.

After making this change, you should commit and push the updated README.md to your repository. This should resolve the issue of the link not working when accessed from the start page of your repository.
